toolkit/offchain/src/reserve/init.rs

```diff
@@ -0,0 +1,265 @@
+//! Initialization of the reserve management is execution of three similar transaction to
+//! initialize three scripts: Rerserve Management Validator, Reserve Management Policy and
+//! Illiquid Circulation Supply Validator.
+//!
+//! Transaction for each of these scripts should have:
+//! * an output to Version Oracle Validator address that should:
+//! * * have script reference with the script being initialized attached, script should be applied with Version Oracle Policy Id
+//! * * contain 1 of token Version Oracle Policy with "Version oracle" asset name, minted in this transaction
+//! * * * mint redeemer should be Constr(1, [Int: SCRIPT_ID, Bytes: Applied Script Bytes])
+//! * * have Plutus Data that is [Int: SCRIPT_ID, Bytes: Version Oracle Policy Id]
+//! * an output to the current governance that should:
+//! * * contain a new Goveranance Policy token, minted in this transaction,
+//! * * * mint redeemer should be empty contructor Plutus Data
+//! * a script reference rnput of the current Goveranance UTXO
+//! * signature of the current goveranance
+
+use crate::{
+	await_tx::AwaitTx,
+	csl::{
+		get_builder_config, get_validator_budgets, zero_ex_units, TransactionBuilderExt,
+		TransactionContext,
+	},
+	init_governance::{get_governance_data, GovernanceData},
+	plutus_script::PlutusScript,
+	scripts_data::{self, VersionOracleData},
+};
+use anyhow::anyhow;
+use cardano_serialization_lib::{
+	AssetName, Assets, BigNum, ConstrPlutusData, DataCost, ExUnits, Int, JsError, LanguageKind,
+	MinOutputAdaCalculator, MintBuilder, MintWitness, MultiAsset, PlutusData, PlutusList,
+	PlutusScriptSource, Redeemer, RedeemerTag, ScriptHash, ScriptRef, Transaction,
+	TransactionBuilder, TransactionOutputBuilder,
+};
+use ogmios_client::{
+	query_ledger_state::{QueryLedgerState, QueryUtxoByUtxoId},
+	query_network::QueryNetwork,
+	transactions::{OgmiosEvaluateTransactionResponse, Transactions},
+};
+use raw_scripts::{ILLIQUID_CIRCULATION_SUPPLY_VALIDATOR, RESERVE_AUTH_POLICY, RESERVE_VALIDATOR};
+use sidechain_domain::{McTxHash, UtxoId};
+use std::collections::HashMap;
+
+pub async fn init_reserve_management<
+	T: QueryLedgerState + Transactions + QueryNetwork + QueryUtxoByUtxoId,
+	A: AwaitTx,
+>(
+	genesis_utxo: UtxoId,
+	payment_key: [u8; 32],
+	client: &T,
+	await_tx: &A,
+) -> anyhow::Result<Vec<McTxHash>> {
+	let reserve_validator =
+		ScriptData::new("Reserve Management Validator", RESERVE_VALIDATOR.to_vec(), 28);
+	let reserve_policy =
+		ScriptData::new("Reserve Management Policy", RESERVE_AUTH_POLICY.to_vec(), 29);
+	let ics_validator = ScriptData::new(
+		"Illiquid Circulation Validator",
+		ILLIQUID_CIRCULATION_SUPPLY_VALIDATOR.to_vec(),
+		30,
+	);
```

Comment:
```suggestion
	let reserve_validator =
		ScriptData::new("Reserve Management Validator", RESERVE_VALIDATOR.to_vec(), raw_scripts::ScriptId::ReserveValidator as u16);
	let reserve_policy =
		ScriptData::new("Reserve Management Policy", RESERVE_AUTH_POLICY.to_vec(), raw_scripts::ScriptId::ReserveAuthPolicy as u16);
	let ics_validator = ScriptData::new(
		"Illiquid Circulation Validator",
		ILLIQUID_CIRCULATION_SUPPLY_VALIDATOR.to_vec(),
		raw_scripts::ScriptId::IlliquidCirculationSupplyValidator as u16,
	);
```

Reply:
Good one, I'll change the type in ScriptData, to reduce casts